### ISSUE TITLE
[Accessibility]: Pie chart has inconsistent focus behaviour 

### ISSUE THREAD

**Issue by vovakulikov** [sourcegraph/sourcegraph#44045]
### Audit type

Keyboard navigation

### Problem description

The current pie chart implementation has inconsistent elements order as you hover over one of the pie arcs. This leads to inconsistent focus order as well (since all pie chart arcs are focusable) 

The element order is corrupted because of the visual effect of hover. We have to pull elements from standard SVG order and put them in as the last element since SVG doesn't support z-index styles over elements. 

### Expected behavior

Hover/focus or any other user input shouldn't change element order and, therefore should do nothing with focus order as well. All visual effects should be implemented with additional hidden from screen reader and keyboard elements. 

### Additional details

If we want to render hover active pie arch on top of over arcs, it makes sense to have two arcs 
- original in origin element place (preserve element order)
- and hidden from the screen reader duplicate in the same position as the original element but in the different (last) position in the DOM 

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [ ] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

/cc @joelkw @felixfbecker @vovakulikov

**Comment by vovakulikov** [comment]
This is P0 since this is a big problem for keyboard and screen reader navigation in Code Insights UI
